### PR TITLE
DM-7261: Port to Python 3

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,6 +15,7 @@ doc/html
 doc/*.tag
 doc/*.inc
 doc/doxygen.conf
+doc/xml
```
Comment (timj) [inline .gitignore]: Also `tests/.cache` I think.

File: examples/makeNoiseCoadd.py [hunk 2]
```diff
@@ -1,30 +1,32 @@
...
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
 # see <http://www.lsstcorp.org/LegalNotices/>.
 #
 
 from __future__ import with_statement
+from builtins import range
 """Make a coadd from gaussian noise images
 """
+from __future__ import print_function
```
Comment (timj) [inline examples/makeNoiseCoadd.py]: We usually put the `__future__` lines before other imports.

File: examples/plotHistogram.py [hunk 3]
```diff
@@ -1,30 +1,32 @@
...
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
 # see <http://www.lsstcorp.org/LegalNotices/>.
 #
 
 from __future__ import with_statement
+from builtins import zip
 """Plot a histogram for a chi squared coadd and overlay a chi squared distribution
 """
+from __future__ import print_function
```
Comment (timj) [inline examples/plotHistogram.py]: Same comment about grouping the `__future__` imports.

File: examples/plotHistogram.py [hunk 4]
```diff
@@ -1,30 +1,32 @@
...
-# 
-# You should have received a copy of the LSST License Statement and 
-# the GNU General Public License along with this program.  If not, 
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
 # see <http://www.lsstcorp.org/LegalNotices/>.
 #
 
 from __future__ import with_statement
```
Comment (timj) [inline examples/plotHistogram.py]: This can be removed. It's not relevant for 2.7.

Comment (r-owen) [inline examples/plotHistogram.py]: Good catch. I found this in several spots.

File: .gitignore (hunk 1 above)
Comment (r-owen) [inline .gitignore]: I should add that. Oddly enough when I run py.test I don't see this; I'm not sure why

File: examples/plotHistogram.py (hunk 3 above)
Comment (r-owen) [inline examples/plotHistogram.py]: I took a manual pass through all imports.

**Pull request merged**